### PR TITLE
INT-1395-cancel-payment

### PR DESCRIPTION
**Pull request by Francois-Gomis**

### CODE REVIEW

**Review by Benjamin-Freoua-Alma — APPROVED** [review]

**Review by syjust-alma — APPROVED** [review]

File: src/Endpoints/Payments.php [hunk 1]
```diff
@@ -119,6 +119,21 @@ public function create($data)
...
+     * @param string $id  The ID of the payment to cancel
+     *
+     * @return void
+     * @throws RequestError
+     */
+    public function cancel($id)
+    {
+        $res = $this->request(self::PAYMENTS_PATH . "/$id/cancel")->put();
+        if ($res->isError()) {
+            $this->logger->error('Error during payment cancellation', [$res]);
```
[the remark this answers is not in the document]
Comment (ClaireAlma) [reply src/Endpoints/Payments.php:132]: Maybe sprinft('An error occurred while canceling the payment id "%s"', $id) is better as message

**Review by ClaireAlma — APPROVED** [review]

File: src/Endpoints/Payments.php [hunk 2]
```diff
@@ -119,6 +119,21 @@ public function create($data)
...
 
+    /**
+     * @param string $id  The ID of the payment to cancel
+     *
+     * @return void
+     * @throws RequestError
+     */
+    public function cancel($id)
+    {
+        $res = $this->request(self::PAYMENTS_PATH . "/$id/cancel")->put();
```
Comment (ClaireAlma) [inline src/Endpoints/Payments.php:130]: I saw everywhere stuff like " self::PAYMENTS_PATH . "/$id/cancel" " but it is not a good practice to build url we shoult use at least sprintf

Comment (Francois-Gomis) [reply src/Endpoints/Payments.php:130]: yes, it is a correction to be expected,everywhere at the same time

File: src/Endpoints/Payments.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (ClaireAlma) [reply src/Endpoints/Payments.php:132]: Ok : , "An error occurred while canceling the payment." is still a better message ;)

**Pull request merged**